### PR TITLE
ThisBuild / versionScheme support + versions 0.3.0

### LINKED ISSUE #26: versionScheme setting
Ref https://github.com/sbt/sbt/issues/5710

Does it make sense for the default use case to use

```scala
ThisBuild / versionScheme := Some(...)
```

as opposed to 

```scala
compatibilityRules += "org.scala-lang.modules" %% "scala-xml" % "..."
```

?

We can derive the `compatibiltyRules` automatically from `projectID`.

### PR DESCRIPTION
**Pull request by eed3si9n**
This is sent on top of https://github.com/scalacenter/sbt-compatibility/pull/31
Fixes #26

This introduces support for `versionScheme` setting. The build user can specify `ThisBuild / versionScheme` to change the semantics of automatic Mima configuration.

### CODE REVIEW

**Comment by julienrf** [comment]
@eed3si9n Do you mind rebasing this one?

**Comment by eed3si9n** [comment]
@julienrf Done.

File: sbt-version-policy/src/sbt-test/sbt-version-policy/mima-semver-spec/test [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+> reload
+
+# OK to apply backward binary compatible changes
+> v1-0-1/versionPolicyCheck
+> v1-0-1/publishLocal
+> reload
+
+# Unable to apply source incompatible changes between
+# two patch versions
+-> v1-0-2/versionPolicyCheck
```
Comment (julienrf) [inline sbt-version-policy/src/sbt-test/sbt-version-policy/mima-semver-spec/test:36]: I thought it would be the case only with `early-semver`?

**Review by julienrf — APPROVED** [review]
This is great, thank you @eed3si9n! I only have one doubt about a test, see below.

File: sbt-version-policy/src/sbt-test/sbt-version-policy/mima-pvp/test [hunk 2]
```diff
@@ -0,0 +1,48 @@
+> v0-1-0/publishLocal
+> reload
```
[the remark this answers is not in the document]
Comment (eed3si9n) [reply sbt-version-policy/src/sbt-test/sbt-version-policy/mima-pvp/test:2]: I think I copied from https://github.com/scalacenter/sbt-version-policy/blob/d61c4abaf05b99f39f7028ad5eb9aaf59f255a75/sbt-version-policy/src/sbt-test/sbt-version-policy/library/test. It's possible that reload there is not necessary.

File: sbt-version-policy/src/sbt-test/sbt-version-policy/mima-semver-spec/test (hunk 1 above)
Comment (eed3si9n) [reply sbt-version-policy/src/sbt-test/sbt-version-policy/mima-semver-spec/test:36]: After `1.0.0`, I would think Early Semver and Semver Spec would behave exactly the same way.

**Pull request merged**